shards/modules/network/network_ws.cpp

```diff
@@ -0,0 +1,703 @@
+#include "network.hpp"
+#include <shards/core/shared.hpp>
+#include <shards/core/params.hpp>
+#include <shards/common_types.hpp>
+#include <shards/core/wire_doppelganger_pool.hpp>
+#include <shards/core/into_wire.hpp>
+#include <shards/core/object_var_util.hpp>
+#include <boost/lockfree/queue.hpp>
+
+#if SH_EMSCRIPTEN
+#include <emscripten/websocket.h>
+#else
+extern "C" {
+#include "pollnet.h"
```

Comment:
hmm what's this? websocket for non emscripten?

Reply:
Yeah C interface for the one on desktop, since it's implemented in rust using an existing library.
Was easier to integrate the common interface that way